### PR TITLE
generate systemd: fix pod dependencies

### PR DESCRIPTION
**Pull request by vrothberg**
Change the dependencies from a pod unit to its associated container units from `Requires` to `Wants` to prevent the entire pod from transitioning to a failed state.  Restart policies for individual containers can be configured separately.

Fixes: #14546
Signed-off-by: Valentin Rothberg <vrothberg@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note
Change the dependencies from a pod to its containers in `generate systemd`  from `Requires=` to `Wants=` to prevent the entire pod from transitioning into the failed state when one container dies.
```

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
@containers/podman-maintainers PTAL

File: pkg/systemd/generate/pods.go [hunk 1]
```diff
@@ -252,6 +252,7 @@ func generatePodInfo(pod *libpod.Pod, options entities.GenerateSystemdOptions) (
 		StopTimeout:       stopTimeout,
 		GenerateTimestamp: true,
 		CreateCommand:     createCommand,
+		RunRoot:           pod.Runtime().RunRoot(),
```
Comment (Luap99) [inline pkg/systemd/generate/pods.go:255]: This looks unrelated? Also you could just use `infraCtr`, so you do not have to add the new
`func (p *Pod) Runtime() *Runtime` function

Comment (vrothberg) [reply pkg/systemd/generate/pods.go:255]: Yes, that's a drive-by fix.  I can repush but it feels like a half-full half-empty thing.

Comment (Luap99) [reply pkg/systemd/generate/pods.go]: The thing here you did not change any tests. It does not seem to part of the template so this is pointless, no?

Comment (vrothberg) [reply pkg/systemd/generate/pods.go]: It's part of the template but not checked for in the e2e tests.  The unit tests set RunRoot.  I can add a test but did not find it worth.

Comment (vrothberg) [reply pkg/systemd/generate/pods.go]: I'll add an e2e test for it :^) Better safe than sorry

**Comment by vrothberg** [comment]
Ready to merge
